bika/lims/api/snapshot.py

```diff
@@ -302,7 +305,7 @@ def take_snapshot(obj, store=True, **kw):
 
     # return immediately
     if not store:
-        return snapshot
+        return data
```

Comment:
Why to return the jsonified dict instead of the dict itself (as it used to be)? If `store' equals `True`, the function will return the dict "as-is". IMO, having a function returning two different types is confusing. 

This brings me the attention to what's the purpose of the function when `store=False`? There is the function `supports_snapshots` that you are already using, like here (you don't call `take_snapshot` unless supported by the object itself):
https://github.com/senaite/senaite.core/blob/1fa2c09c6f09a0c2c11007bdab294e634e758301/bika/lims/subscribers/auditlog.py#L32-L51

Maybe I am wrong, but since the function `supports_snapshots` is already in place and you are taking into consideration, I don't see why this `store True/False` parameter is needed here.